client/src/app/navbar/navbar-app/navbar-app.component.ts

```diff
@@ -1,8 +1,39 @@
 import { Component } from '@angular/core';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { UserModalComponent } from 'src/app/components/user-modal/user-modal/user-modal.component';
+import { UserDTO } from 'src/app/interfaces/user';
+import { UserService } from 'src/app/services/user.service';
 
 @Component({
   selector: 'navbar-app-root',
   templateUrl: 'navbar-app.component.html',
   styleUrls: ['navbar-app.component.scss'],
+  providers: [UserService],
 })
-export class NavbarAppComponent {}
+export class NavbarAppComponent {
+  user: UserDTO | undefined;
+  constructor(
+    private userService: UserService,
+    private ModalService: NgbModal
+  ) {}
+
+  async ngOnInit(): Promise<void> {
+    const user = (await this.userService.getLoggedInUser()).toPromise();
+    user.then((user) => {
+      this.user = user;
+    });
+  }
+
+  editUser() {
+    const modalRef = this.ModalService.open(UserModalComponent, {
+      backdrop: 'static',
+      keyboard: false,
+    });
+    modalRef.componentInstance.user = this.user;
+    modalRef.result.then((result) => {
+      if (result) {
+        location.reload();
```

Comment:
Comment from David for previous PRs:
For temporary solutions, I would recommend making sure you can't accidentally forget to change it later. A code comment would be a good simple solution (otherwise people reading your code in the future have no idea you intend to change it, you can put // TODO: make API call instead of reloading), for important TODOs you can add it to the task tracker.